### PR TITLE
[RLlib] DatasetReader action normalization

### LINKED ISSUE #27335: [RLlib] DatasetReader not normalizing
### What happened + What you expected to happen

There are some problems with the pendulum datasets. I was looking at the min and max values of the actions in the 4 pendulum datasets and this is what I got:
small.json: (-0.977, 0.975)
large.json: (-7810.596, 8995.267)
enormous: (-0.99994, 0.99990)
pendulum_replay_v1.1.0: (-1.999997, 2.0)
So basically pendulum_replay and large is not normalize. Luckily no regression tests use large.json, but I do see that learning_tests_pendulum_crr uses pendulum_replay. However, it does assume that the input is normalized, so I'm not sure how it's not breaking the evaluation, because during eval the env_runner unsquashes the (already unnormalized) actions.

Also there's a flag called `actions_in_input_normalized` which if set to False, JsonReader will normalize the action before passing it to the algorithm. But DatasetReader doesn't.

### Versions / Dependencies

Latest master

### Reproduction script

N/A

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by charlesjsun**
## Why are these changes needed?

DatasetReader didn't normalize actions when `actions_in_input_normalized` was False, which meant CRR was using unnormalized actions during training.

Added normalization to DatasetReader (same as JsonReader)

## Related issue number

Closes #27335

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by avnishn — COMMENTED** [review]
this looks good to me. I think one thing is that if there isn't a test for this already, we should make the test you added apply not just to offline datasets but also to other regular environments. 

If you think im wrong lmk :)

File: rllib/offline/dataset_reader.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import ray.data
 from ray.rllib.offline.input_reader import InputReader
 from ray.rllib.offline.io_context import IOContext
-from ray.rllib.offline.json_reader import from_json_data
+from ray.rllib.offline.json_reader import from_json_data, postprocess_actions
```
Comment (avnishn) [inline rllib/offline/dataset_reader.py:11]: nice

**Comment by charlesjsun** [comment]
> this looks good to me. I think one thing is that if there isn't a test for this already, we should make the test you added apply not just to offline datasets but also to other regular environments.
> 
> If you think im wrong lmk :)

This PR doesn't apply in the online case because this just normalizes actions from dataset (i.e. DatasetReader)

**Review by avnishn — COMMENTED** [review]
The only other thing I would test if I were you is that if you set any other flag combination of actions_in_input_normalized and normalize_actions that the actions won't be normalized by dataset if they aren't supposed to be

**Comment by charlesjsun** [comment]
> Reference

Addressed! Added all combinations of actions_in_input_normalized and normalize_actions

**Review by avnishn — APPROVED** [review]

**Pull request merged**